### PR TITLE
Fix various ProcessTests failures

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/18471
Fixes https://github.com/dotnet/corefx/issues/16958
Fixes https://github.com/dotnet/corefx/issues/16235

### CODE REVIEW

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 1]
```diff
@@ -668,7 +708,9 @@ public void TestPriorityClassWindows()
         public void TestInvalidPriorityClass(ProcessPriorityClass priorityClass)
         {
             var process = new Process();
-            Assert.Throws<InvalidEnumArgumentException>(() => process.PriorityClass = priorityClass);
+            var exception = Record.Exception(() => process.PriorityClass = priorityClass);
```
Comment (danmosemsft) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: how is this different to previous

Comment (ViktorHofer) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: The `Assert.Throws<T>(Action)` method always failed as it said the type is an ArgumentException instead of an InvalidEnumArgumentException. It seems the type resolution is somewhat buggy. With the explicit type check `Assert.IsType<T>(object)` it succeeds.

Comment (Priya91) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: @ViktorHofer This shouldn't repro anymore, this was fixed, please revert this change. All the process tests passed in the latest outerloop [run ](https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/lastCompletedBuild/consoleFull).

```
System.Diagnostics.Process.Tests  Total: 206, Errors: 0, Failed: 0, Skipped: 0, Time: 12.211s
```

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 2]
```diff
@@ -1453,7 +1529,11 @@ public void Process_StartTest()
             Assert.Equal(userName, p.StartInfo.UserName);
             Assert.Same(password, p.StartInfo.Password);
             Assert.Equal(domain, p.StartInfo.Domain);
+
+            // Make sure that process is fully created before killing it to avoid AccessDeniedExceptions.
+            Thread.Sleep(500);
```
Comment (Priya91) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: Instead of thread.sleep, why dont you check if p.Id is > 0. If process hasn't started yet, this will throw InvalidOperationException.

Comment (ViktorHofer) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: > This shouldn't repro anymore, this was fixed, please revert this change. All the process tests passed in the latest outerloop run.

The test TestInvalidPriorityClass failed locally on my machine with current source from master. Should I really revert it?

Comment (Priya91) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: I don't understand how it repros for you, can you post the stack trace

Comment (ViktorHofer) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]:
```
 Discovering: System.Diagnostics.Process.Tests
  Discovered:  System.Diagnostics.Process.Tests
  Starting:    System.Diagnostics.Process.Tests
     System.Diagnostics.Tests.ProcessTests.TestInvalidPriorityClass(priorityClass: 0) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ComponentModel.InvalidEnumArgumentException)
        Actual:   typeof(System.ArgumentException): The value of argument 'value' (0) is invalid for Enum type 'System.
  Diagnostics.ProcessPriorityClass'.
        Stack Trace:
           C:\Users\vikto\Documents\src\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(485,0):
   at System.Diagnostics.Process.set_PriorityClass(ProcessPriorityClass value)
           C:\Users\vikto\Documents\src\corefx\src\System.Diagnostics.Process\tests\ProcessTests.cs(711,0): at System.D
  iagnostics.Tests.ProcessTests.<>c__DisplayClass53_0.<TestInvalidPriorityClass>b__0()
     System.Diagnostics.Tests.ProcessTests.TestInvalidPriorityClass(priorityClass: 96) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ComponentModel.InvalidEnumArgumentException)
        Actual:   typeof(System.ArgumentException): The value of argument 'value' (96) is invalid for Enum type 'System
  .Diagnostics.ProcessPriorityClass'.
        Stack Trace:
           C:\Users\vikto\Documents\src\corefx\src\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(485,0):
   at System.Diagnostics.Process.set_PriorityClass(ProcessPriorityClass value)
           C:\Users\vikto\Documents\src\corefx\src\System.Diagnostics.Process\tests\ProcessTests.cs(711,0): at System.D
  iagnostics.Tests.ProcessTests.<>c__DisplayClass53_0.<TestInvalidPriorityClass>b__0()
  Finished:    System.Diagnostics.Process.Tests
```

I have absolutely no idea... Where was it fixed? In coreclr or corefx or some other dependencies?
